src/v/cluster/topics_frontend.cc

```diff
@@ -1207,43 +1207,50 @@ ss::future<std::error_code> topics_frontend::increase_replication_factor(
                 topic,
                 assignment.id);
               error = errc::topic_invalid_replication_factor;
-              co_return;
+              return ss::now();
           }
 
           partition_constraints.partition_id = assignment.id;
 
           auto ntp = model::ntp(topic.ns, topic.tp, assignment.id);
 
-          auto reallocation = co_await _allocator.invoke_on(
-            partition_allocator::shard,
-            [partition_constraints,
-             assignment = std::move(assignment),
-             ntp = std::move(ntp)](partition_allocator& al) {
-                return al.reallocate_partition(
-                  partition_constraints,
-                  assignment,
-                  get_allocation_domain(ntp));
-            });
-
-          if (!reallocation) {
-              vlog(
-                clusterlog.warn,
-                "attempt to find reallocation for topic {}, partition {} "
-                "failed, error: {}",
-                topic,
-                assignment.id,
-                reallocation.error().message());
-              error = reallocation.error();
-              co_return;
-          }
+          return _allocator
+            .invoke_on(
+              partition_allocator::shard,
+              [partition_constraints,
+               assignment = std::move(assignment),
+               ntp = std::move(ntp)](partition_allocator& al) {
+                  return al.reallocate_partition(
+                    partition_constraints,
+                    assignment,
+                    get_allocation_domain(ntp));
+              })
+            .then([&error,
+                   &units,
+                   &new_assignments,
+                   topic,
+                   p_id = assignment.id](
```

Comment:
this looks like a use-after-move, right? (std::move(assignment) then assignment.id capture)